conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/ServerContexts.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.lib;
+
+import com.google.common.annotations.Beta;
+import io.undertow.server.HttpServerExchange;
+
+@Beta
+public interface ServerContexts {
+
+    /** Returns a new {@link ServerRequestContext} which describes the incoming request. */
+    ServerRequestContext createContext(HttpServerExchange exchange, Endpoint endpoint);
```

Comment:
The `Endpoint` parameter isn't used yet, but it's aligned with the request-context, and we wouldn't want to allow contexts to be created without a known endpoint.